python/cernbox_utils/sharing.py

```diff
@@ -5,11 +5,19 @@
 from .script import get_eos_server_string, get_eos_server
 
 # remove duplicates, preserving order
-def squash(seq):
-    seen = set()
-    seen_add = seen.add
-    return [x for x in seq if not (x in seen or seen_add(x))]
+# use the most permissive acl (otherwise EOS picks the first one)
+def squashAcls(acls):
 
+   seen = {}
+
+   for acl in acls:
+      if (acl.entity, acl.name) in seen:
+         if acl.bits != 'rx': #meaning that this acl is at least as permissive (ie rwx+d)
```

Comment:
Sometimes we have `rx!d` let's make sure we are not skipping the ACLs by hard checking only for `rx`.

Reply:
We do not accept other bits. Those would be marked as wrong bits.

https://github.com/cernbox/cboxshareadmin/blob/master/python/cernbox_utils/cmd_share_admin.py#L233